Lib/test/test_asyncio/test_tasks.py

```diff
@@ -1009,20 +1009,33 @@ def gen():
         self.assertEqual(res, "ok")
 
     def test_wait_for_cancellation_race_condition(self):
-        def gen():
-            yield 0.1
-            yield 0.1
-            yield 0.1
-            yield 0.1
+        async def inner():
+            await asyncio.wait_for(asyncio.sleep(1), timeout=2)
+            return 1
 
-        loop = self.new_test_loop(gen)
+        async def main():
+            result = await asyncio.wait_for(inner(), timeout=1)
+            assert result == 1
 
-        fut = self.new_future(loop)
-        loop.call_later(0.1, fut.set_result, "ok")
-        task = loop.create_task(asyncio.wait_for(fut, timeout=1))
-        loop.call_later(0.1, task.cancel)
-        res = loop.run_until_complete(task)
-        self.assertEqual(res, "ok")
+        asyncio.run(main())
```

Comment:
So, I replaced it with this test which I believe actually reproduces the original problem:
https://bugs.python.org/issue37658

However, this also reveals that the issue is not even fixed as `inner()` would complete successfully, yet the result would not be returned.